packages/protocol/contracts/layer1/based2/libs/LibPropose.sol

```diff
@@ -0,0 +1,136 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+
+import { IInbox as I } from "../IInbox.sol";
+import "./LibValidate.sol";
+import "./LibData.sol";
+import "./LibProver.sol";
+import "./LibCodec.sol";
+
+/// @title LibPropose
+/// @notice Library for processing batch proposals and metadata generation in Taiko protocol
+/// @dev Handles the complete batch proposal workflow including:
+///      - Multi-batch proposal validation and processing
+///      - Batch metadata population with build, propose, and prove sections
+///      - Parent metadata hash verification against evidence
+///      - Prover validation and authentication
+///      - Batch limit enforcement and sequential processing
+/// @custom:security-contact security@taiko.xyz
+library LibPropose {
+    // -------------------------------------------------------------------------
+    // Internal Functions
+    // -------------------------------------------------------------------------
+
+    /// @notice Proposes multiple batches in a single transaction
+    /// @param _access Read/write function pointers for storage access
+    /// @param _config The protocol configuration
+    /// @param _summary The current protocol summary
+    /// @param _batches Array of batches to propose
+    /// @param _evidence Evidence containing parent batch metadata
+    /// @return The updated protocol summary
+    function propose(
+        LibState.Access memory _access,
+        I.Config memory _config,
+        I.Summary memory _summary,
+        I.Batch[] memory _batches,
+        I.BatchProposeMetadataEvidence memory _evidence
+    )
+        internal
+        returns (I.Summary memory)
+    {
+        unchecked {
+            require(_batches.length != 0, EmptyBatchArray());
+
+            // Make sure the lask verified batch is not overwritten by a new batch.
+            // Assuming batchRingBufferSize = 100, right after genesis, we can propose up to 99
+            // batches, the following requirement-statement will pass as:
+            //  1 (nextBatchId) + 99 (_batches.length) <=
+            //  0 (lastVerifiedBatchId) + 100 (batchRingBufferSize)
+            require(
+                _summary.nextBatchId + _batches.length
+                    <= _summary.lastVerifiedBatchId + _config.batchRingBufferSize,
```

Comment:
If we've decided to [remove the support of `forcedInclusion`](https://github.com/taikoxyz/taiko-mono/pull/19734), seems then I can't find a use case where any proposer will need to propose multiple batches in one transaction, a simple batch is enough to utilize all possible blobs? And the proposer will need to pay more `livenessBond` for that. Correct me if I'm wrong...